### PR TITLE
GUM-11: Add create skill form

### PR DESCRIPTION
**Pull request by R-Mielamud**

### CODE REVIEW

File: src/components/form-field/form-field.tsx [hunk 1]
```diff
@@ -1,9 +1,53 @@
...
+
+  const firstError = useMemo<string | null>(
+    () => findFirstError(error),
+    [error, findFirstError]
+  );
+
+  return (
+    <View style={styles.field}>
+      {children}
+      {firstError ? <Text style={styles.error}>{firstError}</Text> : null}
```
Comment (katyakostikova) [inline src/components/form-field/form-field.tsx]:
```suggestion
      {firstError && <Text style={styles.error}>{firstError}</Text>}
```

File: src/components/input/theme.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
+import { DefaultTheme } from '@react-navigation/native';
```
Comment (katyakostikova) [inline src/components/input/theme.ts]:
```suggestion
import { Theme } from 'react-native-paper/lib/typescript/types';
```

File: src/components/input/theme.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
...
+
 const theme = {
```
Comment (katyakostikova) [inline src/components/input/theme.ts]:
```suggestion
const theme: Theme = {
```

File: src/components/input/theme.ts [hunk 4]
```diff
@@ -1,4 +1,7 @@
...
+  ...DefaultTheme,
```
Comment (katyakostikova) [inline src/components/input/theme.ts]:
```suggestion
```

I afraid that this line will override theme that I've done in src/styles/theme

File: src/components/input/select.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  return (
+    <DropDown
+      mode="flat"
+      activeColor={AppColor.ACCENT}
+      visible={open}
+      showDropDown={show}
+      onDismiss={hide}
+      // eslint-ignore-next-line @typescript-eslint/no-explicit-any
+      theme={theme as any}
```
Comment (katyakostikova) [inline src/components/input/select.tsx]:
```suggestion
      theme={theme}
```

File: src/components/form-field/form-field.tsx (hunk 1 above)
Comment (R-Mielamud) [reply src/components/form-field/form-field.tsx]: fixed in [e4ed6df](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/16/commits/e4ed6df525c133157c5109fd79f7db8fbaeb773b)

File: src/components/input/theme.ts (hunk 3 above)
Comment (R-Mielamud) [reply src/components/input/theme.ts]: fixed in [e4ed6df](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/16/commits/e4ed6df525c133157c5109fd79f7db8fbaeb773b)

File: src/components/input/select.tsx (hunk 5 above)
Comment (R-Mielamud) [reply src/components/input/select.tsx]: fixed in [e4ed6df](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/16/commits/e4ed6df525c133157c5109fd79f7db8fbaeb773b)

File: src/components/input/theme.ts (hunk 4 above)
Comment (R-Mielamud) [reply src/components/input/theme.ts]: fixed in [c4e646a](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/16/commits/c4e646a8c21dbfdbaa44c8abd63e2b557e432645)

File: src/components/input/theme.ts (hunk 2 above)
Comment (R-Mielamud) [reply src/components/input/theme.ts]: fixed in [e4ed6df](https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup-mobile/pull/16/commits/e4ed6df525c133157c5109fd79f7db8fbaeb773b)

**Review by katyakostikova — APPROVED** [review]

File: src/common/enums/skill/skill-type.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+enum SkillType { // TODO: real values from server
+  SoftSkills,
+  HardSkills,
+  Other,
+}
```
Comment (liza-veis) [inline src/common/enums/skill/skill-type.ts]:
```suggestion
enum SkillType { 
  SOFT_SKILLS = 'SoftSkills',
  HARD_SKILLS = 'HardSkills',
  OTHER = 'Other',
}
```

File: src/common/helpers/enum/get-numeric-enum-values.helper.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+type Enum = Record<string | number, string | number>;
```
Comment (liza-veis) [inline src/common/helpers/enum/get-numeric-enum-values.helper.ts]: Remove this helper

File: src/common/types/skill/create-skill.interface.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { SkillType } from 'src/common/enums';
+
+interface ICreateSkill {
```
Comment (liza-veis) [inline src/common/types/skill/create-skill.interface.ts]:
```suggestion
interface ICreateSkillPayload {
```

File: src/components/form-field/form-field.tsx [hunk 9]
```diff
@@ -1,9 +1,53 @@
...
+import { Text } from '..';
 import styles from './styles';
 
-const FormField: React.FC = ({ children }) => {
-  return <View style={styles.field}>{children}</View>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Error = string | string[] | FormikErrors<any> | FormikErrors<any>[];
+
+interface Props {
+  error?: Error;
```
Comment (liza-veis) [inline src/components/form-field/form-field.tsx]:
```suggestion
  error?: string;
```

File: src/components/form-field/form-field.tsx [hunk 10]
```diff
@@ -1,9 +1,53 @@
...
+
+  const firstError = useMemo<string | null>(
+    () => findFirstError(error),
+    [error, findFirstError]
+  );
+
+  return (
+    <View style={styles.field}>
+      {children}
+      {firstError && <Text style={styles.error}>{firstError}</Text>}
```
Comment (liza-veis) [inline src/components/form-field/form-field.tsx]:
```suggestion
      {Bollean(error) && <Text style={styles.error}>{error}</Text>}
```

File: src/components/form-field/form-field.tsx [hunk 11]
```diff
@@ -1,9 +1,53 @@
...
+
+      return findFirstError(Object.values(currentError) as string[]);
+    },
+    [touched]
+  );
+
+  const firstError = useMemo<string | null>(
+    () => findFirstError(error),
+    [error, findFirstError]
+  );
```
Comment (liza-veis) [inline src/components/form-field/form-field.tsx]: Remove this logic, if we set values types, we won't need it

File: src/components/input/select.tsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import DropDown from 'react-native-paper-dropdown';
+import { AppColor } from 'src/common/enums';
+
+import styles from './styles';
```
Comment (liza-veis) [inline src/components/input/select.tsx]:
```suggestion

import { AppColor } from 'src/common/enums';
import styles from './styles';
```

File: src/components/input/select.tsx [hunk 13]
```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
```
Comment (liza-veis) [inline src/components/input/select.tsx]: Move it to a separate component components/select

File: src/components/input/select.tsx [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+  React.ComponentPropsWithoutRef<typeof DropDown>,
+  'visible' | 'onDismiss' | 'showDropDown'
+>;
+
+const Select: React.FC<Props> = dropDownProps => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  const show = () => setOpen(true);
+  const hide = () => setOpen(false);
+
```
Comment (liza-veis) [inline src/components/input/select.tsx]: Component methods must be named via handle

```suggestion
  const handleShow = () => setOpen(true);
  
  const handleHide = () => setOpen(false);

```

File: src/screens/create-skill/create-skill.tsx [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+import { ICreateSkill } from 'src/common/types';
+import { useAppDispatch } from 'src/hooks';
+import { skillActions } from 'src/store/actions';
+import CreateSkillForm from './form';
+
+const CreateSkill: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation();
+
+  const back = () => {
```
Comment (liza-veis) [inline src/screens/create-skill/create-skill.tsx]:
```suggestion
  const handleGoBack = () => {
```

File: src/screens/create-skill/create-skill.tsx [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+    navigation.goBack();
+  };
+
+  const submit = (data: ICreateSkill) => {
```
Comment (liza-veis) [inline src/screens/create-skill/create-skill.tsx]:
```suggestion
  const handleSubmit = (data: ICreateSkill) => {
```

File: src/screens/create-skill/create-skill.tsx [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    dispatch(skillActions.create(data));
+    back();
+  };
+
+  return (
+    <SafeAreaView>
+      <CreateSkillForm onSubmit={submit} onCancel={back} />
```
Comment (liza-veis) [inline src/screens/create-skill/create-skill.tsx]:
```suggestion
      <CreateSkillForm onSubmit={submit} onBackPress={back} />
```

File: src/screens/create-skill/form.tsx [hunk 18]
```diff
@@ -0,0 +1,140 @@
+import React from 'react';
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]: Move this to create-skill/components/create-skill-form.tsx and follow other components file structure

File: src/screens/create-skill/form.tsx [hunk 19]
```diff
@@ -0,0 +1,140 @@
...
+
+import { getNumericEnumValues } from 'src/common/helpers';
+import styles from './styles';
+
+interface Props {
+  onSubmit: (data: ICreateSkill) => void;
+  onCancel: () => void;
+}
+
+const skillTypeOptions = [
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]: Move it to create-skill/constants.ts

File: src/screens/create-skill/form.tsx [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+  Heading,
+  Input,
+  MainButton,
+  Select,
+} from 'src/components';
+
+import { getNumericEnumValues } from 'src/common/helpers';
+import styles from './styles';
+
+interface Props {
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]:
```suggestion
type ICreateSkillScreenProps {
```

File: src/screens/create-skill/form.tsx [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+    label: 'Soft skills',
+    value: SkillType.SoftSkills,
+  },
+  {
+    label: 'Other',
+    value: SkillType.Other,
+  },
+];
+
+const validationSchema = yup.object({
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]: Move this to src/validation-schemas/skill/create-skill

File: src/screens/create-skill/form.tsx [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+    .typeError('Estimate must be a valid number')
+    .min(0, 'Estimate must be from 0 to 10')
+    .max(10, 'Estimate must be from 0 to 10')
+    .required('Estimate must not be empty'),
+});
+
+const CreateSkillForm: React.FC<Props> = ({ onSubmit, onCancel }) => {
+  return (
+    <Form<ICreateSkill>
+      initialValues={{ type: SkillType.Other }}
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]: Create DEFAULT_CREATE_SKILL_PAYLOAD : ICreateSkillPayload, add all initial values and put in src/constants.ts

File: src/screens/create-skill/form.tsx [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+    >
+      {({
+        handleChange,
+        handleSubmit,
+        setFieldValue,
+        values,
+        errors,
+        touched,
+        isValid,
+      }) => (
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]:
```suggestion
      }: ICreateSkillPayload ) => (
```

File: src/screens/create-skill/form.tsx [hunk 24]
```diff
@@ -0,0 +1,140 @@
...
+    .required('Estimate must not be empty'),
+});
+
+const CreateSkillForm: React.FC<Props> = ({ onSubmit, onCancel }) => {
+  return (
+    <Form<ICreateSkill>
+      initialValues={{ type: SkillType.Other }}
+      validationSchema={validationSchema}
+      validateOnChange={true}
+      validateOnBlur={true}
```
Comment (Liza-Veis) [inline src/screens/create-skill/form.tsx:62]: Remove this

File: src/screens/create-skill/form.tsx [hunk 25]
```diff
@@ -0,0 +1,140 @@
...
+        values,
+        errors,
+        touched,
+        isValid,
+      }) => (
+        <View>
+          <Heading level={HeadingLevel.H4}>Create skill</Heading>
+          <FormField
+            error={errors.type}
+            touched={touched.type as boolean | undefined}
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]:
```suggestion
            touched={touched.type}
```

File: src/screens/profile/profile.tsx [hunk 26]
```diff
@@ -1,7 +1,9 @@
 import React, { useState } from 'react';
 import { View } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
 import { FAB } from 'react-native-paper';
 import { SafeAreaView } from 'react-native-safe-area-context';
+import { ProfileRoute } from 'src/common/enums';
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]:
```suggestion

import { ProfileRoute } from 'src/common/enums';
```

File: src/screens/profile/profile.tsx [hunk 27]
```diff
@@ -10,27 +12,38 @@ import styles from './styles';
...
-    location: () => ...,
-    education: () => ...,
-    language: () => ...,
-    careerPoint: () => ...,
-    interest: () => ...,
+    skill: () =>
+      navigation.navigate({
+        name: ProfileRoute.CREATE_SKILL as never,
+        params: {} as never,
+      }),
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]:
```suggestion
    skill: () => {
      navigation.navigate({
        name: ProfileRoute.CREATE_SKILL as never,
        params: {} as never,
      })
   },
```

File: src/screens/create-skill/styles.ts [hunk 28]
```diff
@@ -0,0 +1,10 @@
+import { StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  buttons: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
```
Comment (liza-veis) [inline src/screens/create-skill/styles.ts]: Check what buttons look like if we switch between screens
Screenshot 2022-02-14 at 10 35 47

File: src/screens/profile/profile.tsx [hunk 29]
```diff
@@ -10,27 +12,38 @@ import styles from './styles';
...
+    interest: () => {
+      /* TODO */
+    },
   };
-  */
 
   const handleItemPress = (name?: string) => {
     if (!name) {
       return;
     }
```
Comment (liza-veis) [inline src/screens/profile/profile.tsx]:
```suggestion
  const handleItemPress = (name: string) => {
```

File: src/services/skillApi/index.ts [hunk 30]
```diff
@@ -0,0 +1,28 @@
+import { ApiPath, ContentType, HttpMethod } from 'src/common/enums';
```
Comment (liza-veis) [inline src/services/skillApi/index.ts]: Please rename this file to skill-api and also authApi to auth-api

File: src/store/skill/actions.ts [hunk 31]
```diff
@@ -0,0 +1,14 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { ICreateSkill } from 'src/common/types';
```
Comment (liza-veis) [inline src/store/skill/actions.ts]:
```suggestion
import { createAsyncThunk } from '@reduxjs/toolkit';

import { ICreateSkill } from 'src/common/types';
```

File: src/store/skill/actions.ts [hunk 32]
```diff
@@ -0,0 +1,14 @@
...
+import { skillApi } from 'src/services';
+import { ActionTypes } from './common';
+
+const create = createAsyncThunk(
```
Comment (liza-veis) [inline src/store/skill/actions.ts]:
```suggestion
const createSkill = createAsyncThunk(
```

File: src/store/skill/common.ts [hunk 33]
```diff
@@ -0,0 +1,5 @@
+enum ActionTypes {
+  Create = 'skill/create',
```
Comment (liza-veis) [inline src/store/skill/common.ts]:
```suggestion
  CREATE_SKILL = 'skill/create',
```

File: src/store/skill/slice.ts [hunk 34]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+import { create } from './actions';
+
+const initialState = {};
+
+const slice = createSlice({
+  name: 'skills',
```
Comment (liza-veis) [inline src/store/skill/slice.ts]:
```suggestion
  name: 'skill',
```

**Review by liza-veis — DISMISSED** [review]

File: src/store/actions.ts [hunk 35]
```diff
@@ -1 +1,2 @@
 export { authActions } from './auth';
+export { skillActions } from './skill';
```
Comment (liza-veis) [inline src/store/actions.ts]: Let's export all actions together
```suggestion
export * from './auth';
export * from './skill';
```

File: src/screens/create-skill/form.tsx [hunk 36]
```diff
@@ -0,0 +1,140 @@
...
+    label: 'Other',
+    value: SkillType.Other,
+  },
+];
+
+const validationSchema = yup.object({
+  type: yup
+    .number()
+    .integer()
+    .oneOf(getNumericEnumValues(SkillType))
```
Comment (liza-veis) [inline src/screens/create-skill/form.tsx]:
```suggestion
    .oneOf(Object.values(SkillType))
```

**Comment by yevhenprybytok** [comment]
Let's change FormField component into a component that can control form props (such as touched, error etc.) on by itself. So we can use at as `<FormField name='usename' component={InputComponent} />`. Also we need to define props that will not change for our form (like validationSchema, initialValues), so we always know what to expect and we can control what can be passed through the props into the Formik component.

File: src/screens/create-skill/form.tsx (hunk 36 above)
Comment (R-Mielamud) [reply src/screens/create-skill/form.tsx:44]: Знімок екрана 2022-02-14 о 15 07 56

Comment (R-Mielamud) [reply src/screens/create-skill/form.tsx:44]: This won't work

File: src/screens/create-skill/form.tsx (hunk 25 above)
Comment (R-Mielamud) [reply src/screens/create-skill/form.tsx]: Знімок екрана 2022-02-14 о 15 11 57

Comment (R-Mielamud) [reply src/screens/create-skill/form.tsx]: Won't work

File: src/screens/create-skill/form.tsx (hunk 23 above)
Comment (R-Mielamud) [reply src/screens/create-skill/form.tsx]: This neither is a ICreateSkillPayload nor returns ICreateSkillPayload

File: src/screens/create-skill/form.tsx (hunk 36 above)
Comment (liza-veis) [reply src/screens/create-skill/form.tsx]: To make it work you need to rewrite enum and change validation schema type to string
```
enum SkillType { 
  SOFT_SKILLS = 'SoftSkills',
  HARD_SKILLS = 'HardSkills',
  OTHER = 'Other',
}
```
Screenshot 2022-02-14 at 15 15 29

**Comment by R-Mielamud** [comment]
SkillType is an unsigned integer in the database

**Comment by R-Mielamud** [comment]
Select has the same theme and styles. Better to place it near to input, than copy styles and theme.

**Comment by R-Mielamud** [comment]
> Also we need to define props that will not change for our form (like validationSchema, initialValues), so we always know what to expect and we can control what can be passed through the props into the Formik component.

This `React.ComponentPropsWithoutRef<typeof (Something)>` is common for all `src/components`.

File: src/screens/create-skill/form.tsx (hunk 23 above)
Comment (liza-veis) [reply src/screens/create-skill/form.tsx]: Yeah, you are right it's my mistake

**Review by yevhenprybytok — APPROVED** [review]

File: src/common/enums/skill/skill-type.ts [hunk 37]
```diff
@@ -0,0 +1,7 @@
+enum SkillType { // TODO: real values from server
+  SOFT_SKILLS,
+  HARD_SKILLS,
+  OTHER,
+}
```
Comment (liza-veis) [inline src/common/enums/skill/skill-type.ts]: And also remove helper for numeric enums
```suggestion
enum SkillType { // TODO: real values from server
  SOFT_SKILLS = 'softSkills',
  HARD_SKILLS = 'hardSkills',
  OTHER = 'other',
}
```

File: src/components/form/form-input/form-input.tsx [hunk 38]
```diff
@@ -17,7 +17,7 @@ const FormInput: React.FC<FormInputProps> = ({
   containerStyle,
   ...inputProps
 }) => {
-  const { values, errors, touched, handleChange, handleBlur } =
+  const { values, errors, touched, setFieldValue, handleBlur } =
     useFormikContext<FormikValues>();
```
Comment (liza-veis) [inline src/components/form/form-input/form-input.tsx:22]:
```suggestion
const handleChange = (value: string) => setFieldValue(name, value);
```

File: src/components/form/form-input/form-input.tsx [hunk 39]
```diff
@@ -26,7 +26,7 @@ const FormInput: React.FC<FormInputProps> = ({
     <View style={containerStyle}>
       <Input
         value={values[name]}
-        onChangeText={handleChange(name)}
+        onChangeText={text => setFieldValue(name, text)}
```
Comment (liza-veis) [inline src/components/form/form-input/form-input.tsx]:
```suggestion
        onChangeText={handleChange}
```

File: src/components/form/form-select/form-select.tsx [hunk 40]
```diff
@@ -0,0 +1,37 @@
...
+  const { values, errors, touched, setFieldValue } =
+    useFormikContext<FormikValues>();
+
+  const error = touched[name] && errors[name];
+
+  return (
+    <View style={containerStyle}>
+      <Select
+        value={values[name]}
+        setValue={val => setFieldValue(name, val)}
```
Comment (liza-veis) [inline src/components/form/form-select/form-select.tsx]: Same here, create handleChange callback

File: src/components/select/select.tsx [hunk 41]
```diff
@@ -0,0 +1,33 @@
...
+type Props = Omit<
+  React.ComponentPropsWithoutRef<typeof DropDown>,
+  'visible' | 'onDismiss' | 'showDropDown'
+>;
+
+const Select: React.FC<Props> = dropDownProps => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  const handleShow = () => setOpen(true);
+  const handleHide = () => setOpen(false);
```
Comment (liza-veis) [inline src/components/select/select.tsx]:
```suggestion
  const handleShow = () => setOpen(true);
  
  const handleHide = () => setOpen(false);
```

File: src/screens/create-skill/components/create-skill-form/create-skill-form.tsx [hunk 42]
```diff
@@ -0,0 +1,77 @@
...
+
+import { ButtonMode, HeadingLevel } from 'src/common/enums';
+import { ICreateSkillPayload } from 'src/common/types/skill';
+import { createSkillValidationSchema } from 'src/validation-schemas';
+import { FormInput, FormSelect, Heading, MainButton } from 'src/components';
+
+import {
+  DEFAULT_CREATE_SKILL_PAYLOAD,
+  SKILL_TYPE_OPTIONS,
+} from '../../constants';
```
Comment (liza-veis) [inline src/screens/create-skill/components/create-skill-form/create-skill-form.tsx]: Rename constants file to common and also name variables there in camel case (we discussed it in the group today)

File: src/screens/create-skill/components/create-skill-form/create-skill-form.tsx [hunk 43]
```diff
@@ -0,0 +1,77 @@
...
+import styles from '../../styles';
+
+interface Props {
+  onSubmit: (data: ICreateSkillPayload) => void;
+  onBack: () => void;
+}
+
+const CreateSkillForm: React.FC<Props> = ({ onSubmit, onBack }) => {
+  return (
+    <Formik<ICreateSkillPayload>
```
Comment (liza-veis) [inline src/screens/create-skill/components/create-skill-form/create-skill-form.tsx]: We don't need this type if we set initialValues
```suggestion
    <Formik
```

File: src/screens/create-skill/components/create-skill-form/create-skill-form.tsx [hunk 44]
```diff
@@ -0,0 +1,77 @@
...
+      initialValues={DEFAULT_CREATE_SKILL_PAYLOAD}
+      validationSchema={createSkillValidationSchema}
+      validateOnMount={true}
+      validateOnChange={true}
+      onSubmit={values =>
+        onSubmit({ ...values, estimate: Number(values.estimate) })
+      }
```
Comment (liza-veis) [inline src/screens/create-skill/components/create-skill-form/create-skill-form.tsx]: Let's move it to callback like handleCreateSkill or handleFormSubmit

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: src/screens/create-skill/components/create-skill-form/create-skill-form.tsx [hunk 45]
```diff
@@ -0,0 +1,77 @@
...
+  onSubmit: (data: ICreateSkillPayload) => void;
+  onBack: () => void;
+}
+
+const CreateSkillForm: React.FC<Props> = ({ onSubmit, onBack }) => {
+  return (
+    <Formik<ICreateSkillPayload>
+      initialValues={DEFAULT_CREATE_SKILL_PAYLOAD}
+      validationSchema={createSkillValidationSchema}
+      validateOnMount={true}
```
Comment (liza-veis) [inline src/screens/create-skill/components/create-skill-form/create-skill-form.tsx:26]: Do we need to validate on mount?

File: src/screens/create-skill/components/create-skill-form/create-skill-form.tsx [hunk 46]
```diff
@@ -0,0 +1,77 @@
...
+      validateOnChange={true}
```
Comment (liza-veis) [inline src/screens/create-skill/components/create-skill-form/create-skill-form.tsx]: validateOnChange set to true by default

**Review by liza-veis — APPROVED** [review]

**Pull request merged**